src/fromager/commands/lint_requirements.py

```diff
@@ -47,6 +62,23 @@ def lint_requirements(input_files_path: list[pathlib.Path]) -> None:
                     raise InvalidRequirement(
                         "Constraints files cannot contain extra dependencies"
                     )
+
+                # Resolve the requirement to ensure it can be found
+                # Skip resolution for constraints files as they should only specify versions
+                if not path.name.endswith("constraints.txt"):
+                    try:
+                        token = requirement_ctxvar.set(requirement)
+                        _, version = bt.resolve_version(
+                            req=requirement,
+                            req_type=RequirementType.TOP_LEVEL,
+                        )
+                        logger.info(f"{requirement} resolves to {version}")
+                        requirement_ctxvar.reset(token)
+                    except Exception as resolve_err:
+                        logger.error(
+                            f"{path}: {line}: Failed to resolve requirement: {resolve_err}"
+                        )
+                        flag = False
```

Comment:
The context var was not reset in error case.

```suggestion
                    except Exception as resolve_err:
                        logger.error(
                            f"{path}: {line}: Failed to resolve requirement: {resolve_err}"
                        )
                        flag = False
                    finally:
                        requirement_ctxvar.reset(token)
```

Reply:
Nice, fixed.